hrcase/CRM/HRCase/Upgrader.php

```diff
@@ -116,14 +116,26 @@ public function upgrade_1402() {
     $this->up1402_removeUnusedCaseTypes();
     //Removes CiviCase activity types which should belong to CiviTask component
     $this->removeActivityTypesList($defaultActivityTypes, 'CiviCase');
-    $this->up1402_createDefaultCaseTypes($defaultCaseTypes);
+    $this->createOrUpdateDefaultCaseTypes($defaultCaseTypes);
     $this->up1402_createDefaultActivityTypes($defaultActivityTypes);
     $this->changeActivityTypeComponent('Open Case', 'CiviCase', 'CiviTask');
 
     return TRUE;
   }
 
   /**
+   * Resets all default case types discarding any customization to match new
+   * activity workflow
+   */
+  public function upgrade_1404() {
+    $this->installActivityTypes('CiviDocument', ['P45']);
```

Comment:
you could have used up1402_createDefaultActivityTypes and changed it to createOrUpdateDefaultActivityType similar to what you did with createOrUpdateDefaultCaseTypes .

so the whole thing would be : 

1) inside civihr/hrcase/CRM/HRCase/DefaultCaseAndActivityTypes.php , change $defaultActivityTypes  to the following :

```php
  private static $defaultActivityTypes = [
    'Schedule Exit Interview' => 'CiviTask',
    'Get "No Dues" certification' => 'CiviTask',
    'Conduct Exit Interview' => 'CiviTask',
    'Revoke Access to Database' => 'CiviTask',
    'Block work email ID' => 'CiviTask',
    'Background Check' => 'CiviTask',
    'References Check' => 'CiviTask',
    'Schedule joining date' => 'CiviTask',
    'Issue appointment letter' => 'CiviTask',
    'Fill Employee Details Form' => 'CiviTask',
    'Submission of ID/Residence proofs and photos' => 'CiviTask',
    'Program and work induction by program supervisor' => 'CiviTask',
    'Enter employee data in CiviHR' => 'CiviTask',
    'Group Orientation to organization, values, policies' => 'CiviTask',
    'Probation appraisal (start probation workflow)' => 'CiviTask',
    'Confirm End of Probation Date' => 'CiviTask',
    'Start Probation workflow' => 'CiviTask'
    'P45' =>'CiviDocument',
  ];
```

and here change up1402_createDefaultActivityTypes  to :

```php
  private function createOrUpdateActivityTypes($defaultActivityTypes) {
    foreach ($defaultActivityTypes as $activityType => $componentName) {
      $type = civicrm_api3('OptionValue', 'get', [
        'sequential' => 1,
        'name' => $activityType,
        'option_group_id' => 'activity_type',
        'component_id' =>  $componentName,
      ]);

      if (empty($type['id'])) {
        civicrm_api3('OptionValue', 'create', [
          'sequential' => 1,
          'option_group_id' => 'activity_type',
          'label' => $activityType,
          'name' => $activityType,
          'component_id' =>  $componentName,
        ]);
      }
    }
  }
```